main/api.c

```
@@ -346,6 +384,12 @@ static void listen_cb(evutil_socket_t sock, short what, void * /*priv*/) {
 		return;
 	}
 
+	/* check if peer is root or has CAP_NET_ADMIN */
+	if (check_permission(fd) != 0) {
+		close(fd);
+		return;
```

For consistency with the rest of the code:

```c
if (check_client_permissions(fd) < 0) {
```

Can we return an explicit error to the client?

---

need to dig into grout to understand API msg. I will check that.